### PR TITLE
[Updated] Improved Contacts Manger Flow in Settings

### PR DESCRIPTION
**Pull request by BboyAkers**
Fixes: #7069

Explanation:
This Flow is based upon the updated UI Figma Flow
https://www.figma.com/file/giaffk6qfl9eMMjoPFjJkGnt/Address-book?node-id=816%3A6600

Manual testing steps:

    Settings -> Add A New Contact

### CODE REVIEW

**Comment by darkwing** [comment]
Wow @BboyAkers, this is an awesome start!  Functionally this piece is awesome!  A few items:

- We can remove the placeholders from all of the inputs, per the design mockups
- The labels should match the Figma mockup (i.e. "Username" -> "Contact name", "Ethereum Public Address" -> "E... p... a....")
- Would it be possible to match button styles, or would it effect all modals?

**Comment by BboyAkers** [comment]
* Definiltey!
* I'll change that 🙂 
* It'll only effect this modal. I'll change the button styles. I thought part of the design was out of date, considering many other buttons in the app have more of a rounded property to them. Does having more of a box-like design signify we're in a modal?

File: ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js
Comment (darkwing) [inline ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js:42]: I think this can be `if (nextProps?.qrCodeData?.type === 'address') {`

File: ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js
Comment (darkwing) [inline ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js:136]: We'll want to localize this.

File: ui/pages/send/send-content/add-recipient/ens-input.component.js
Comment (darkwing) [inline ui/pages/send/send-content/add-recipient/ens-input.component.js:217]: We could probably do: `placeholderText || t('recipientAddressPlaceholder')`

**Comment by darkwing** [comment]
@rachelcope Do we want the edits to be done in that modal as well?

**Comment by BboyAkers** [comment]
Hey! I've updated the modal. My only problem is the text overflow in memo. Would someone mind taking a look at that? @danjm @darkwing @danfinlay @rekmarks

**Comment by darkwing** [comment]
Hello @BboyAkers !  Found a few items:

1.  Based on the Figma mockup, It looks like the first heading should be `Contact name`
2. Could we `autofocus` the first form field so the user can start typing immediately?
3. I think we can include in this PR the updated screen for when you click on a contact in full screen:

SidebySide

**Comment by BboyAkers** [comment]
Added autofocus, and updated the edit button(s). I did ask awhile ago about the name. I was told it should be username 🙂

**Comment by BboyAkers** [comment]
@darkwing Hey the e2e tests keep failing can someone help me with them? I'm also almost ready to create another PR for some feature work!

**Comment by darkwing** [comment]
@PeterYinusa Could help @BboyAkers on this, as well as do a bit of QA?  I've been so lost in 1559 that I've not been able to help Austin on this as much as I wanted.

**Comment by PeterYinusa** [comment]
Hey @BboyAkers, this is looking great so far! Can you have a look at the following items:

### ENS resolution
Steps:
1. Click Add Contact
2. Enter an ENS name in the Ethereum Public Address field

Actual result:
invalid address
Invalid address

Expected result:
The ens name resolves to an address on supported networks

### Failed to update error
Steps:
1. Click Add Contact
2. Enter an invalid address in the Ethereum Public Address field
3. Click Save
4. Click Advanced menu item

Actual result:
Address book failed to update
Address book failed to update

Expected result:
Save button is disabled if the address is invalid

### Styling
Steps:
Expanded view
1. Click Add Contact

Actual result:
Add Memo placeholder is styled different in comparison to Username and Ethereum Public Address input fields
Placeholder styling

Expected result:
Consistent styling

### Long usernames
Steps:
Popup view
1. Click Add Contact
2. Enter 'this is a very long user name' in the Username  field
2. Enter a valid address
3. Click Save
4. Select the newly created contact

Actual result:
Back arrow and close button is pushed out the view
Username continues behind the Add Contact button
Long username

Expected result:
The username is truncated with an ellipses similar to the address

**Comment by BboyAkers** [comment]
@peterjan I've fixed what you've logged except for the ens Input. I think lavamoat isn't allowing me to send an http request to validate the ens input. Mind taking a look? 🙂

File: ui/components/app/modals/modal.js
Comment (brad-decker) [inline ui/components/app/modals/modal.js:144]: We should not be using the Modal component or adding new modals to this older Modal system. Instead we have a <Popover> component that is used in a number of places. Modal is planned for deprecation.

File: ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js
Comment (brad-decker) [inline ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js:16]: [REQ] in addition to moving this from modals and using a popover, all new components we create should not use the .component / .container breakdown and instead use functional components that use the useSelector/useDispatch hooks to tie into redux stores

File: ui/pages/settings/contact-list-tab/contact-list-tab.component.js
Comment (brad-decker) [inline ui/pages/settings/contact-list-tab/contact-list-tab.component.js:38]: [REQ] We should maintain only one component per file. As this is a separate react component, we would need to move it to its own file. Although I'm not sure much is gained by separating out this component from the JSX tree. Maybe just render the button inline instead of making it separate?

File: ui/pages/settings/contact-list-tab/contact-list-tab.component.js
Comment (brad-decker) [inline ui/pages/settings/contact-list-tab/contact-list-tab.component.js:29]: When the add new contact modal is a popover, it can be rendered inline and only displayed when a state variable changes. So instead of having to dispatch a showModal, you can just set a state variable that gets passed to the Popover. This will help keep more of the logic around adding a contact in roughly the same place. It's also fine to separate them into different components.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Hey @BboyAkers :D thanks for continuing to refine this. I have a few best practice/code design alignment things on this

**Comment by BboyAkers** [comment]
I'll get crackin on these changes and will have a PR up by Monday 🙂

File: ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js
Comment (BboyAkers) [inline ui/components/app/modals/add-new-contact-modal/add-new-contact-modal.component.js:56]: @brad-decker Can we use lookUpEnsName from ducks/ens.js to replace this line? 🙂

**Review by mcmire — CHANGES_REQUESTED** [review]
Hi @BboyAkers, thanks for this PR. I know this has been batted around a bit but I did a first pass and have some comments of my own. Most are minor, but there are a couple of files where you access the environment type and I have a hunch that they may cause problems. I have yet to pull down your branch and do a manual test on this, so perhaps it is fine, but I just wanted to note that for now at least.

**Comment by BboyAkers** [comment]
@mcmire I'll make these changes tonight 🙂

**Comment by bschorchit** [comment]
Action item: I should review this with designers.

**Comment by brad-decker** [comment]
@BboyAkers @bschorchit is this work still in progress and needed? If so can we get this rebased and ready for another review.

**Comment by BboyAkers** [comment]
I definitely need to get this rebased and ready for another review!

**Comment by brad-decker** [comment]
@bschorchit @BboyAkers,

First off, thank you so much for the work you have done so far on this. We greatly appreciate it! We should talk about having one of our team members take this over the finish line to make sure your contributions are landed. As you can see the branch is out of date and conflicts have arisen, and that's fine! At the same time, the longer this goes the more conflicts will occur and the more work will be needed to get it to a mergeable/reviewable state. I know you're contributing your free time to making MetaMask better, and we <3 you for it. I have been trying to get on top of our PR queue and i've been closing internal team members PRs that have gone stale in addition to external contributors, so me asking if its the right time to switch owners on this PR is not a reflection of anything other than I would love to see this landed and our PR queue is crazy and I want to help focus reviewers time. 

If you do **want** to do this work yourself, then by all means but I want to provide you with support from our team as well if it would make more sense.

**Comment by brad-decker** [comment]
No, thank you for doing the work in the first place. We appreciate it immensely. @bschorchit @hilvmason @kevinghim heads up we need to have someone dedicated to helping get this landed in the near future before it goes out of date. given that there is already some feedback I dont' believe it to be more than a full day or two effort from a dedicated engineer.

**Comment by brad-decker** [comment]
Hey there @BboyAkers,

We have recently landed some refreshed UI for the contacts page, and its going into a release very soon. Thank you for your contribution and work done here! Going to close this as the designs now differ from the target of these changes.

**Comment by darkwing** [comment]
Thank you again @BboyAkers for your hard work and efforts on this!  I apologize for it getting stale!  Let's catch up soon!

**Pull request closed without merging**
